.gitlab-ci.yml

```
@@ -10,27 +11,37 @@ stages:
 
 variables:
   REPO_LANG: python # "python" is used everywhere rather than "py"
-  TESTRUNNER_IMAGE: registry.ddbuild.io/images/mirror/dd-trace-py/testrunner:ecc5741ff3e7c8a30363fcd9cca79a371dcea5b4
   # CI_DEBUG_SERVICES: "true"
 
-.testrunner:
-  image: $TESTRUNNER_IMAGE
-  # DEV: we have a larger pool of amd64 runners, prefer that over arm64
-  tags: [ "arch:amd64" ]
-  timeout: 20m
-  before_script:
-    - pyenv global 3.12 3.7 3.8 3.9 3.10 3.11 3.13-dev
-    - export _CI_DD_AGENT_URL=http://${HOST_IP}:8126/
-
 include:
   - remote: https://gitlab-templates.ddbuild.io/libdatadog/include/one-pipeline.yml
   - local: ".gitlab/services.yml" # Include early so others can use the definitions
   - local: ".gitlab/benchmarks.yml"
   - local: ".gitlab/package.yml"
-  - local: ".gitlab/tests.yml"
   - local: ".gitlab/macrobenchmarks.yml"
   - local: ".gitlab/dogfood.yml"
   - local: ".gitlab/release.yml"
+  - local: ".gitlab/testrunner.yml"
+
+tests-gen:
+  stage: tests-gen
+  extends: .testrunner
+  script:
+    - pip install riot~=0.19.1
```

We're using 0.20.0 pinned in dd-trace-py since [chore(ci): bump riot version for 3.7 compatibility issues](https://github.com/DataDog/dd-trace-py/pull/11085)
```suggestion
    - pip install riot==0.20.0
```